packages/client/src/comp/states/Base.tsx

```diff
@@ -33,8 +64,19 @@ export const Base = <Info extends StateInfo>({ enrichValues, children }: Props<I
   const { contact } = ContactContainer.useContainer()
   const { voter } = VoterContainer.useContainer()
   const { fetchingData, setFetchingData } = FetchingDataContainer.useContainer()
+  const [ registrationStatus, setRegistrationStatus ] = React.useState<ExtendedStatus>(null)
+
+  const [ validFirstName, setValidFirstName ] = React.useState(false)
```

Comment:
Consider using `Unstated` and make `valid: Record<InputId, boolean>` and `setValid(inputId: InputId, value: boolean)`.